### ISSUE TITLE
zizmor check: pages workflow

### ISSUE THREAD

**Issue by funnelfiasco** [guacsec/guac-docs#175]
```
warning[artipacked]: credential persistence through GitHub Actions artifacts
  --> .github/workflows/pages.yml:32:9
   |
32 |         - name: Checkout
   |  _________-
33 | |         uses: actions/checkout@v3
   | |_________________________________- does not set persist-credentials: false
   |

error[excessive-permissions]: overly broad workflow or job-level permissions
  --> .github/workflows/pages.yml:17:1
   |
17 | / permissions:
18 | |   contents: read
...  |
21 | | 
22 | | # Allow one concurrent deployment
   | |_________________________________^ pages: write is overly broad at the workflow level
   |

error[excessive-permissions]: overly broad workflow or job-level permissions
  --> .github/workflows/pages.yml:17:1
   |
17 | / permissions:
18 | |   contents: read
...  |
21 | | 
22 | | # Allow one concurrent deployment
   | |_________________________________^ id-token: write is overly broad at the workflow level
   |

info[template-injection]: code injection via template expansion
  --> .github/workflows/pages.yml:43:9
   |
43 |       - name: Build with Jekyll
   |         ----------------------- info: this step
44 |         # Outputs to the './_site' directory by default
45 |         run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}"
   |         ------------------------------------------------------------------------------ info: steps.pages.outputs.base_path may expand into attacker-controllable code
   |

4 findings (0 unknown, 1 informational, 0 low, 1 medium, 2 high)
```

**Comment by funnelfiasco** [comment]
Failed with
```
[Invalid workflow file: .github/workflows/pages.yml#L62](https://github.com/guacsec/guac-docs/actions/runs/11685358163/workflow)
The workflow is not valid. .github/workflows/pages.yml (Line: 62, Col: 9): Unexpected value 'permissions'
```

I'll follow up with another PR

**Comment by mihaimaruseac** [comment]
Really awesome that you ran this! Thank you

**Comment by funnelfiasco** [comment]
Fixed in #175 and fixed the fix in #177
